components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m

```diff
@@ -182,10 +182,16 @@ - (void)updateTitleFont {
   if (!_titleFont) {
     _titleFont = [MDCFeatureHighlightView defaultTitleFont];
   }
-  if (_mdc_adjustsFontForContentSizeCategory) {
-    _titleLabel.font =
-        [_titleFont mdc_fontSizedForMaterialTextStyle:kTitleTextStyle
-                                 scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
+  if (_mdc_adjustsFontForContentSizeCategory && !_mdc_legacyFontScaling) {
+    if (_titleFont.mdc_scalingCurve) {
```

Comment:
Does `mdc_scalingCurve` need to be checked here?

As a developer, I thought the logic could be 
`
if (__mdc_adjustsFontForContentSizeCategory) {
  if (! _mdc_legacyFontScaling) {
    _titleLabel.font = [_titleFont mdc_scaledFontForCurrentSizeCategory]; // no matter the curve exists or not
  } else {
   _titleLabel.font =
          [_titleFont mdc_fontSizedForMaterialTextStyle:kTitleTextStyle
                                   scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
  }
} else {
  _titleLabel.font = _titleFont;
}
`
I might think in the wrong way. Please feel free to correct me.

Reply:
You are absolutely correct.
I moved the !_mdc_legacyFontScaling to the proper clause, but I kept the .mdcScalingCurve check because it adds context to the dependent clause.
Fixed.